soc/arm/ene_kb1200/soc.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023 ENE Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/arch/cpu.h>
+#include <soc.h>
+
+static void pmu_init(void)
+{
+	PMU_T *pmu = ((PMU_T *)PMU_BASE);
+	/* Interrupt Event Wakeup from IDLE mode Enable */
+	pmu->PMUIDLE |= 0x00000001;
+	/* GPTD wake up from STOP mode enable. */
+	pmu->PMUSTOP |= PMU_STOP_WU_GPTD;
+	/* SWD EDI32 wake up from STOP mode enable */
+	pmu->PMUSTOP |= (PMU_STOP_WU_EDI32 | PMU_STOP_WU_SWD);
+}
+
+#define KB1200_CLKCFG 0x40000020
+
+static void clock_init(void)
+{
+	if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 96000000) {
+		/* AHB/APB clock select 96MHz/48MHz */
+		sys_write32(0x00000004, KB1200_CLKCFG);
+	} else if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 48000000) {
+		/* AHB/APB clock select 48MHz/24MHz */
+		sys_write32(0x00000014, KB1200_CLKCFG);
+	} else {
+		/* AHB/APB clock select 24MHz/12MHz */
+		sys_write32(0x00000024, KB1200_CLKCFG);
```

Comment:
magic numbers again. this code is also inconsistent: above you're using struct-based access and here sys_write32. please pick one or the other.